wasm/type.go

```diff
@@ -11,6 +11,30 @@ type FunctionType struct {
 	Params, Results []ValueType
 }
 
+var nullary = []byte{0x60, 0, 0}
+
+var encodedOneParam = buildEncodedFuncTypes(true)
+var encodedOneResult = buildEncodedFuncTypes(false)
+
+// buildEncodedFuncTypes is like buildEncodedValTypes except it is parameter or result specific and includes the func
+// type prefix 0x60.
+//
+// See https://www.w3.org/TR/wasm-core-1/#function-types%E2%91%A4
+func buildEncodedFuncTypes(param bool) (encodedTypes [ValueTypeI32 + 1][]byte) {
```

Comment:
`[ValueTypeI32 + 1][]byte` seems too big and redundant because `ValueTypeI32 = 0x7f`?

Reply:
maybe `ValueTypeF64 - ValueTypeI32`

Reply:
yeah we can make it smaller, but then to access the map entry, we have subtract first

Reply:
at this point probably just as well to use a switch statement.